### PR TITLE
Add permission filter to program listing endpoint 

### PR DESCRIPTION
**Pull request by kdmccormick**
@edx/masters-neem 
[EDUCATOR-4348](https://openedx.atlassian.net/browse/EDUCATOR-4348)

### CODE REVIEW

File: registrar/apps/core/utils.py [hunk 1]
```diff
@@ -57,3 +57,44 @@ def serialize_to_csv(items, field_names, include_headers=False):
     for item in items:
         writer.writerow(item)
     return outfile.getvalue()
+
+
+def memoize_as(memo_var_name):
```
Comment (schenedx) [inline registrar/apps/core/utils.py]: Can you explain the difference between this decorator, and `cached_property` decorator done at https://docs.djangoproject.com/en/2.2/ref/utils/?

**Review by schenedx — COMMENTED** [review]
I took a rough look and had a question. Good work overall

File: registrar/apps/core/utils.py (hunk 1 above)
Comment (kdmccormick) [reply registrar/apps/core/utils.py]: Nothing, I wish I knew that `cached_property` existed a long time ago!

Comment (kdmccormick) [reply registrar/apps/core/utils.py]: I'll update it to use Django's version :)

**Review by jansenk — APPROVED** [review]
This looks good to me :)

**Pull request merged**